### PR TITLE
IPC-124: Show same price to customer on re-quote

### PR DESCRIPTION
**Pull request by mightyost**
# Jira Issue: [IPC-124] 

## What?
If tariffs are changed and the customer creates a new quote then the price should be reused from old quote given that  any previous (equal) quote exists within 30 days.

## Why?
- Anton wants this

## Optional checklist
- [ ] Codescouted
- [x] Unit tests written
- [x] Tested locally

[IPC-124]: https://hedvig.atlassian.net/browse/IPC-124

### CODE REVIEW

File: src/main/kotlin/com/hedvig/underwriter/service/RequotingService.kt [hunk 1]
```diff
@@ -91,9 +145,151 @@ class RequotingService(
...
+    }
+
     private fun QuoteInitiatedFrom.isAnyOf(vararg initiatedFrom: QuoteInitiatedFrom): Boolean =
         initiatedFrom.any { this == it }
 
     private fun AgreementStatus.isAnyOf(vararg statuses: AgreementStatus): Boolean =
         statuses.any { this == it }
 }
+
+fun MonetaryAmount.isSameAmount(amount: BigDecimal?): Boolean {
```
Comment (AlexSJ1990) [inline src/main/kotlin/com/hedvig/underwriter/service/RequotingService.kt:179]: nice!

File: src/main/kotlin/com/hedvig/underwriter/service/RequotingService.kt [hunk 2]
```diff
@@ -91,9 +145,151 @@ class RequotingService(
...
+        }
+
+        val props1 = obj1.asMap()
+        val props2 = obj2.asMap()
+
+        for ((field, value1) in props1) {
+
+            val value2 = props2[field]
+
+            val same = comparators[field]?.invoke(value1, value2) ?: true
```
Comment (Michael-Duivestein) [inline src/main/kotlin/com/hedvig/underwriter/service/RequotingService.kt:245]: Nice!

File: src/main/kotlin/com/hedvig/underwriter/service/RequotingService.kt (hunk 1 above)
Comment (mightyost) [reply src/main/kotlin/com/hedvig/underwriter/service/RequotingService.kt]: :)

File: src/main/kotlin/com/hedvig/underwriter/service/RequotingService.kt (hunk 2 above)
Comment (mightyost) [reply src/main/kotlin/com/hedvig/underwriter/service/RequotingService.kt]: Thx!

**Review by AlexSJ1990 — APPROVED** [review]

File: src/main/kotlin/com/hedvig/underwriter/service/RequotingService.kt [hunk 3]
```diff
@@ -91,9 +145,151 @@ class RequotingService(
...
+    return this.with(Monetary.getDefaultRounding()) == other.with(Monetary.getDefaultRounding())
+}
+
+fun Quote.isSame(quote: Quote) =
+    QuoteComparator.isSame(this, quote)
+
+object QuoteComparator {
+
+    val justToMakeSureNoComparatorsAreForgottenWhenAddingNewTypes: (QuoteData) -> String = fun (type: QuoteData): String =
+        when (type) {
```
[the remark this answers is not in the document]
Comment (mightyost) [reply src/main/kotlin/com/hedvig/underwriter/service/RequotingService.kt]: Sorry, forgot to commit it ;).

**Review by fredriklagerblad — APPROVED** [review]
Nice !

**Pull request merged**